android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/SerpTelemetryFetcher.kt

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.telemetry
+
+import android.content.Context
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.android.org.json.toList
+import mozilla.components.support.remotesettings.RemoteSettingsClient
+import mozilla.components.support.remotesettings.RemoteSettingsResult
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.File
+
+internal const val REMOTE_ENDPOINT_URL = "https://firefox.settings.services.mozilla.com"
+internal const val REMOTE_ENDPOINT_BUCKET_NAME = "main"
+
+/**
+ * Parse SERP Telemetry json from remote config.
+ */
+
+class SerpTelemetryFetcher(
+    context: Context,
+    private val collectionName: String,
+    private val serverUrl: String = REMOTE_ENDPOINT_URL,
+    private val bucketName: String = REMOTE_ENDPOINT_BUCKET_NAME,
+    private val rootStorageDirectory: File = context.filesDir,
```

Comment:
I don't think this default parameter is worth the cost of passing `Context` down the object hierarchy. Since `Context` is a god object, I generally try to avoid it as a dependency as much as possible.

I think instead we can just have `private val rootStorageDirectory: File` and initialize the top-level objects like:

`AdsTelemetry(context.filesDir)` and `InContentTelemetry(context.filesDir)`

Reply:
Have been wanting to do this. Thanks for reminding